src/ConnectedMode.UnitTests/Binding/ImportsBeforeFileGeneratorTests.cs

```diff
@@ -0,0 +1,103 @@
+
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.Setup(x => x.Directory.Exists(pathToDirectory)).Returns(true);
+            fileSystem.Setup(x => x.File.Exists(pathToFile)).Returns(false);
+
+            _ = new ImportBeforeFileGenerator(fileSystem.Object);
+
+            fileSystem.Verify(x => x.File.WriteAllText(pathToFile, expectedResult), Times.Once);
+        }
+
+        [TestMethod]
+        public void FileExists_DifferentText_CreatesFile()
+        {
+            string pathToDirectory = GetPathToImportBefore();
+            string pathToFile = Path.Combine(pathToDirectory, "SonarLint.targets");
+            string expectedResult = "<Project xmlns=\"http://schemas.microsoft.com/developer/msbuild/2003\">\r\n  <ItemGroup />\r\n</Project>";
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.Setup(x => x.Directory.Exists(pathToDirectory)).Returns(true);
+            fileSystem.Setup(x => x.File.Exists(pathToFile)).Returns(true);
+            fileSystem.Setup(x => x.File.ReadAllText(pathToFile)).Returns("wrong text");
+
+            _ = new ImportBeforeFileGenerator(fileSystem.Object);
+
+            fileSystem.Verify(x => x.File.WriteAllText(pathToFile, expectedResult), Times.Once);
+        }
+
+        [TestMethod]
+        public void FileExists_SameText_DoesNotCreateFile()
+        {
+            string pathToDirectory = GetPathToImportBefore();
+            string pathToFile = Path.Combine(pathToDirectory, "SonarLint.targets");
+            string expectedResult = "<Project xmlns=\"http://schemas.microsoft.com/developer/msbuild/2003\">\r\n  <ItemGroup />\r\n</Project>";
+
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.Setup(x => x.Directory.Exists(pathToDirectory)).Returns(true);
+            fileSystem.Setup(x => x.File.Exists(pathToFile)).Returns(true);
+            fileSystem.Setup(x => x.File.ReadAllText(pathToFile)).Returns(expectedResult);
+
+            _ = new ImportBeforeFileGenerator(fileSystem.Object);
+
+            fileSystem.Verify(x => x.File.WriteAllText(pathToFile, expectedResult), Times.Never);
+        }
+
+        [TestMethod]
+        public void DirectoryDoesNotExist_CreatesDirectory()
+        {
+            string pathToDirectory = GetPathToImportBefore();
+            string pathToFile = Path.Combine(pathToDirectory, "SonarLint.targets");
+
+            var fileSystem = new Mock<IFileSystem>();
+            fileSystem.Setup(x => x.Directory.Exists(pathToDirectory)).Returns(false);
+            fileSystem.Setup(x => x.File.Exists(pathToFile)).Returns(false);
+
+            _ = new ImportBeforeFileGenerator(fileSystem.Object);
+
+            fileSystem.Verify(x => x.Directory.CreateDirectory(pathToDirectory), Times.Once);
+        }
+
+        private string GetPathToImportBefore()
+        {
+            var localAppData = Environment.GetEnvironmentVariable("LOCALAPPDATA");
```

Comment:
FYI we have an abstraction `IEnvironmentVariableProvider` that allows us to mock environment variables.

I think it's ok not to use it here since we're only using one environment variable, and it's guaranteed to exist on Windows machines. The abstraction is more useful if we are using "custom" environment variables,  or if the expected behaviour of the code changes depending on the value of the variable.